### PR TITLE
Handle multiple experiments with the same name

### PR DESCRIPTION
**Pull request by ngondalia**
Create panel after server instantiates the experiment and tie the panel name to the experiment name

Fixes the constant switching of tabs when multiple experiments opened with the same name

Signed-off-by: Neel Gondalia <ngondalia@blackberry.com>

### CODE REVIEW

**Comment by ngondalia** [comment]
> light2.mp4

This is the issue that this commit addresses

File: vscode-trace-extension/src/trace-explorer/trace-utils.ts [hunk 1]
```diff
@@ -114,29 +107,29 @@ export const fileHandler =
...
-                        traceLogger.addLogMessage('Failed to open trace: ' + traceName, fileHandler.name);
-                        traceLogger.addLogMessage(
-                            'There may be an issue with the server or the trace is invalid.',
-                            fileHandler.name
+                        traceLogger.showError(
+                            'Failed to open trace: ' +
+                                traceName +
+                                '. There may be an issue with the server or the trace is invalid.'
                         );
+                        return;
```
Comment (bhufmann) [inline vscode-trace-extension/src/trace-explorer/trace-utils.ts:115]: If returning here no experiment is created with a subset of traces. The idea was that for a list of traces pathes in tracesArray, each trace is posted to the server. If not successful, it continues with the next trace in the array. If one or more are successfully posted, the experiment is created with these traces. Below there is a warning message created to the user of failed traces. If no trace was successfully posted the no experiment is created. 

I think we should remove the return here

**Review by bhufmann — CHANGES_REQUESTED** [review]
The change makes sense to open the panel with the correct name. 

I was not able to reproduce the flickering. I guess it's timing related. I got hopeful that it would fix the flickering reported in #238, but this update doesn't address it. The Trace Compass team will investigate that issue in a separate effort.

File: vscode-trace-extension/src/trace-explorer/trace-utils.ts (hunk 1 above)
Comment (ngondalia) [reply vscode-trace-extension/src/trace-explorer/trace-utils.ts]: My mistake - fixed!

**Comment by ngondalia** [comment]
> I was not able to reproduce the flickering. I guess it's timing related. I got hopeful that it would fix the flickering reported in #238, but this update doesn't address it. The Trace Compass team will investigate that issue in a separate effort.

It doesnt always happen - it depends on what you click after opening the second trace. For me, its always reproducible if i click the newly opened trace (2nd trace with same name) in Opened Traces view.

**Comment by bhufmann** [comment]
> > I was not able to reproduce the flickering. I guess it's timing related. I got hopeful that it would fix the flickering reported in #238, but this update doesn't address it. The Trace Compass team will investigate that issue in a separate effort.
> 
> It doesnt always happen - it depends on what you click after opening the second trace. For me, its always reproducible if i click the newly opened trace (2nd trace with same name) in Opened Traces view.

I was able to reproduce it now. Thanks for the steps.

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**